codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ServerUpgradeCodecTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpVersion;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import io.netty.util.CharsetUtil;
+import org.junit.Test;
+
+public class Http2ServerUpgradeCodecTest {
+
+    @Test
+    public void testUpgradeToHttp2ConnectionHandler() {
+        testUpgrade(new Http2ConnectionHandlerBuilder().frameListener(new Http2FrameAdapter())
+                .httpClearTextUpgrade(true).build());
+    }
+
+    @Test
+    public void testUpgradeToHttp2FrameCodec() {
+        testUpgrade(new Http2FrameCodecBuilder(true).httpClearTextUpgrade(true).build());
+    }
+
+    @Test
+    public void testUpgradeToHttp2MultiplexCodec() {
+        testUpgrade(new Http2MultiplexCodecBuilder(true, new HttpInboundHandler()).httpClearTextUpgrade(true).build());
+    }
+
+    private static void testUpgrade(Http2ConnectionHandler handler) {
+        FullHttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/");
```

Comment:
Optional: if you treat you test code as examples of how to use the SUT, consider using the http OPTIONS method, and the `*` path.